test/jdk/tools/jpackage/helpers/jdk/jpackage/test/MacHelper.java

```diff
@@ -81,12 +81,23 @@ public static void withExplodedDmg(JPackageCommand cmd,
             ThrowingConsumer<Path, ? extends Exception> consumer) {
         cmd.verifyIsOfType(PackageType.MAC_DMG);
 
+        // Mount DMG under random temporary folder to avoid collisions when
+        // mounting DMG with same name asynchroniusly multiple times.
+        // See JDK-8373105. "hdiutil" does not handle such cases very good.
+        final Path mountRoot;
+        try {
+            mountRoot = Files.createTempDirectory("mountRoot");
```

Comment:
Temporary test files should be created in the test working directory. Use `TKit.createTempDirectory("mount-root")` instead.

It also makes the newly added `TKit.deleteDirectoryRecursive(mountRoot, "");` call redundant, as the test working directory will be deleted automatically when the test exits.

`TKit.createTempDirectory()` throws unchecked exceptions; the `try/catch` is redundant.